libp2p.nimble

```diff
@@ -94,3 +94,47 @@ task examples_build, "Build the samples":
   buildSample("helloworld", true)
   buildTutorial("examples/tutorial_1_connect.md")
   buildTutorial("examples/tutorial_2_customproto.md")
+
+# pin system
+# while nimble lockfile
+# isn't available
+import strscans
+import algorithm
+const PinFile = ".pinned"
+task pin, "Create a lockfile":
+  rmDir("nimbledeps")
+  mkDir("nimbledeps")
+  exec("nimble install -dy")
+  var allDeps = newSeq[string]()
+  for dependency in listDirs("nimbledeps/pkgs"):
+    #echo dependency & "/nimblemeta.json"
+    let fileContent = readFile(dependency & "/nimblemeta.json")
+    var
+      prefix = ""
+      url = ""
+      version = ""
+    discard scanf(fileContent, "$*url\":\"$+\"", prefix, url)
+    discard scanf(fileContent, "$*vcsRevision\":\"$+\"", prefix, version)
+    #if version.len == 0:
+    #  echo execCmdEx("ls")
```

Comment:
In some unknown case (like with dnsclient), the version will not be in the nimblemeta.json file.
I would like to do `git ls-remote https://github.com/ba0f3/dnsclient.nim` and capture the output to get a pinned version, but I couldn't find a way to read the stdout of a command in nimscript.

In the meantime, I put the missing ones manually in the .pinned file, which is not ideal, but works

Reply:
Fixed by switching the pinner to a .nim file